### PR TITLE
Locale specific numbers

### LINKED ISSUE #3410: Add locale aware formatting to numbers
On some of our pages numbers are reaching into the thousand and million count which makes the numbers harder to read without locale specific formatting, as can be seen in this navbar on the front page of the staging server:

screen shot 2014-12-06 at 04 07 20

We should add locale aware formatting that would change e.g.

1000 -> 1,000 or 1 000 or 1.000
1000000 -> 1,000,000 or 1 000 000 or 1.000.000

In a locale aware away so that it is easy to read.

### PR DESCRIPTION
**Pull request by dwaynebailey**
This adds a custom babel based CLDR templatetag and makes use of it for all numbers on Pootle.

Fixes #3410

### CODE REVIEW

File: pootle/apps/pootle_misc/templatetags/common_tags.py [hunk 1]
```diff
@@ -36,14 +37,20 @@ def avatar(username, email_hash, size):
...
-        translated=round(
-            (float(translated)/total) * 100),
-        untranslated=round(
-            (float(untranslated)/total) * 100))
+        untranslated=cldrpercent(untranslated, cldrformat),
+        fuzzy=cldrpercent(fuzzy, cldrformat),
+        translated=cldrpercent(translated, cldrformat),
+        bar_u=round(untranslated * 100),
+        bar_f=round(fuzzy * 100),
+        bar_t=round(translated * 100))
```
Comment (phlax) [inline pootle/apps/pootle_misc/templatetags/common_tags.py]: nothing is returned here - also untranslated, fuzzy etc are explicit - whereas bar_x doesnt mean anything

File: pootle/apps/pootle_misc/templatetags/babel.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from babel import core as babel_core
+
+from django import template
+from django.utils.translation import get_language, to_locale
+
+
+register = template.Library()
+
+
+def _get_format():
```
Comment (phlax) [inline pootle/apps/pootle_misc/templatetags/babel.py]: can we lose the _get_format function, it makes it harder to follow

File: pootle/templates/admin/dashboard.html [hunk 3]
```diff
@@ -11,15 +11,15 @@
...
         </tr>
         <tr>
           <th scope="row">{% trans "Pending suggestions" %}</th>
-          <td class="stats-number">{{ server_stats.pending_count }}</td>
+          <td class="stats-number">{{ server_stats.pending_count|cldrnumber }}</td>
         </tr>
         <tr>
           <th scope="row">{% trans "Users" %}</th>
-          <td class="stats-number">{{ server_stats.user_count }}</td>
+          <td class="stats-number">{{ server_stats.user_count|cldrnumber }}</td>
```
Comment (phlax) [inline pootle/templates/admin/dashboard.html]: i think this probs wants to happen the view display or somewhere else

File: pootle/apps/pootle_misc/templatetags/common_tags.py (hunk 1 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_misc/templatetags/common_tags.py]: I look to rename, but its was getting noisy.

I'm not sure what "nothing is returned here means"

CLDR functions need a fraction ie 0.956 to create 96,6%.  While the bar needs a perecent expressed as 95.6%

**Comment by unho** [comment]
Tested and reviewed. gtm

**Pull request merged**